packages/snippets/bin/snippets.dart

```diff
@@ -41,10 +68,27 @@ const List<String> sampleTypes = <String>[
   'dartpad',
 ];
 
+// This is a hack to workaround the fact that git status inexplicably fails
+// (with random non-zero error code) about 2% of the time.
+String getChannelNameWithRetries() {
+  int retryCount = 0;
+  while(retryCount < 2) {
+    try {
+      return getChannelName();
+    } on GitStatusFailed catch (e) {
+      retryCount += 1;
+      stderr.write('git status failed, retrying ($retryCount)\nError report:\n$e');
+    }
+  }
+  return getChannelName();
+}
+
 /// Generates snippet dartdoc output for a given input, and creates any sample
 /// applications needed by the snippet.
 void main(List<String> argList) {
-  final Map<String, String> environment = Platform.environment;
+  stderr.writeln('Invoked as: snippets ${argList.join(' ')}');
```

Comment:
Was this just for debugging? Or is this useful for production?

Reply:
Oh! Yes, that should be removed.